src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -307,8 +308,8 @@ def update_extension(cmd, extension_name, index_url=None, pip_extra_index_urls=N
         raise CLIError(e)
 
 
-def list_available_extensions(index_url=None, show_details=False):
-    index_data = get_index_extensions(index_url=index_url)
+def list_available_extensions(cmd, index_url=None, show_details=False):
```

Comment:
I think if we only need `cmd.cli_ctx`, we can pass `cmd.cli_ctx` as early as possible.